recipe-box-backend/src/main/java/com/endava/recipebox/service/impl/RecipeServiceImpl.java

```diff
@@ -37,22 +43,58 @@ public static boolean isPublic(Recipe recipe) {
     public List<RecipeDTO> getAllPublicRecipes() {
         return recipeMapper.map(recipeRepository.findAll().stream()
                 .filter(RecipeServiceImpl::isPublic)
-                .collect(Collectors.toList()));
+                .toList());
     }
 
     @Override
     public List<RecipeDTO> getAllPublicRecipesByName(String recipeName) {
         return  recipeMapper.map(recipeRepository.findAll().stream()
                 .filter(RecipeServiceImpl::isPublic)
                 .filter(r -> r.getName().equals(recipeName))
-                .collect(Collectors.toList()));
+                .toList());
     }
 
     @Override
     public List<RecipeDTO> getAllPublicRecipesByType(MealType mealType) {
         return recipeMapper.map(recipeRepository.findAll().stream()
                 .filter(RecipeServiceImpl::isPublic)
                 .filter(r -> r.getMealType() == mealType)
-                .collect(Collectors.toList()));
+                .toList());
+    }
+
+    @Override
+    @Transactional
+    public Recipe createRecipe(RecipeRequestDTO recipeRequestDTO, Long userId) {
+
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new RuntimeException("User not found"));
+        if (user.getRole() != Role.Admin )
+            throw new RuntimeException("Only admins can create recipes");
```

Comment:
Throwing a RuntimeException directly is generally not considered a best practice in a well-structured application, especially in a Spring Boot application where there are better ways to handle errors and provide meaningful responses to clients.
1. Use a Custom Exception
public class UnauthorizedActionException extends RuntimeException {
    public UnauthorizedActionException(String message) {
        super(message);
    }
}
2. Then, modify your code to throw this custom exception throw new UnauthorizedActionException("Only admins can create recipes");
}
3. You can annotate your custom exception with @ResponseStatus to automatically return an HTTP status code when the exception is thrown: @ResponseStatus(HttpStatus.FORBIDDEN)
public class UnauthorizedActionException extends RuntimeException {...}   when the exception is thrown, Spring will return a 403 Forbidden HTTP status code along with the error message.
4. You can centralize your exception handling using @ControllerAdvice. This way, you can manage how exceptions are handled across your entire application:
@ControllerAdvice
public class GlobalExceptionHandler {

    @ExceptionHandler(UnauthorizedActionException.class)
    public ResponseEntity<String> handleUnauthorizedActionException(UnauthorizedActionException ex) {
         ErrorResponse errorResponse = new ErrorResponse(ex.getMessage(), HttpStatus.FORBIDDEN.value());
         return new ResponseEntity<>(errorResponse, HttpStatus.FORBIDDEN);
    }

    // Other exception handlers can be added here
}

4. create the custom error response class: public class ErrorResponse {
    private String message;
    private int statusCode;

    public ErrorResponse(String message, int statusCode) {
        this.message = message;
        this.statusCode = statusCode;
    }

    // Getters and setters
}